src/base.ts

```diff
@@ -0,0 +1,271 @@
+  const outOfDiv = document.createElement('div');
+  const pageCountDiv = document.createElement('div');
+  const nextButton = document.createElement('button');
+  nextButton.className = 'nextButton';
+  const prevButton = document.createElement('button');
+  prevButton.className = 'prevButton';
+  const zoomButton = document.createElement('button');
+  zoomButton.className = 'zoomButton';
+  const loadingTask = getDocument(documentUrl);
+
+  // handle updating page number on scroll
+  canvasContainer.onscroll = () => {
+    const pageNumber = getPage(canvasContainer.scrollTop);
+    pageNumberDiv.textContent = `${pageNumber}`;
+    if (!controlsDiv.style.opacity) {
+      controlsDiv.style.opacity = '1';
+      setTimeout(() => controlsDiv.removeAttribute('style'), 1000);
+    }
+  };
+
+  try {
+    const pdfDocument: PDFDocumentProxy = await loadingTask.promise;
+
+    // initial viewer setup
+    pageNumberDiv.textContent = '1';
+    pageDiv.textContent = 'Page ';
+    outOfDiv.textContent = '/';
+    pageCountDiv.textContent = `${pdfDocument.numPages}`;
+    fullPageNumberDiv.appendChild(pageDiv);
+    fullPageNumberDiv.appendChild(pageNumberDiv);
+    fullPageNumberDiv.appendChild(outOfDiv);
+    fullPageNumberDiv.appendChild(pageCountDiv);
+
+    nextButton.onclick = () => {
+      const pageNumber = parseInt(pageNumberDiv.textContent || '1');
+      if (pageNumber < pdfDocument.numPages) {
+        const nextPage = getPageId(pageNumber + 1);
+        canvasContainer.scrollTo({ top: (document.getElementById(nextPage)?.parentElement as HTMLElement).offsetTop });
+        pageNumberDiv.textContent = `${pageNumber + 1}`;
+      }
+    };
+    prevButton.onclick = () => {
+      const pageNumber = parseInt(pageNumberDiv.textContent || '1');
+      if (pageNumber > 1) {
+        const prevPage = getPageId(pageNumber - 1);
+        canvasContainer.scrollTo({ top: (document.getElementById(prevPage)?.parentElement as HTMLElement).offsetTop });
+        pageNumberDiv.textContent = `${pageNumber - 1}`;
+      }
+    };
+    nextButton.innerHTML = chevronRight;
+    prevButton.innerHTML = chevronLeft;
+
+    zoomButton.innerHTML = plusSvg;
+    containerDiv.appendChild(controlsDiv);
+    controlsDiv.appendChild(prevButton);
+    controlsDiv.appendChild(fullPageNumberDiv);
+    controlsDiv.appendChild(nextButton);
+    controlsDiv.appendChild(zoomButton);
+    containerDiv.appendChild(canvasContainer);
+    const pages = [...Array(pdfDocument.numPages).keys()];
+
+    // load each page synchronously so that they appear in order
+    while (pages.length) {
+      const page = pages.shift();
+      if (page !== undefined) {
+        // render the PDF page
+        const pageContainer = document.createElement('div');
+        pageContainer.className = 'pageContainer';
+        pageContainer.style.width = defaultWidth;
+        canvasContainer.appendChild(pageContainer);
+        const pdfPage = await pdfDocument.getPage(page + 1);
```

Comment:
More awaits.

Reply:
these definitely need to be synchronous -- otherwise the pages load out of order

Reply:
actually let me think about this -- I think we can load the pages simultaneously but still await the `Promise.all`